src/storage/storage_handler.py

```diff
@@ -0,0 +1,52 @@
+from abc import ABC, abstractmethod
+
+from src.utils.config import ConfigArguments
+
+class DataStorage(ABC):
+    def __init__(self):
+        self._args = ConfigArguments.get_instance()
+    
+    @abstractmethod
+    def get_uri(self, file_name):
+        """
+            This method returns URI of a file name based on the implemented file system.
+            eg: For a file in S3, s3:// has to be prefixed to the file name.
+            eg: For a file in hdfs, hdfs:// has to be prefixed to the file name.
+        """
+        pass
+
+    @abstractmethod
+    def makedirs(self, dir_name, exist_ok=False):
```

Comment:
we should name it more generic. The general term is a namespace. Even in filesystem, directories represent a namespace for a file. i would prefer using create_namespace() API.